core/field_number.js

```diff
@@ -41,35 +41,46 @@ goog.require('Blockly.utils.object');
  * @param {Function=} opt_validator A function that is called to validate
  *    changes to the field's value. Takes in a number & returns a validated
  *    number, or null to abort the change.
+ * @param {Object=} opt_config A map of options used to configure the field.
+ *    See the [field creation documentation]{@link https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/number#creation}
+ *    for a list of properties this parameter supports.
  * @extends {Blockly.FieldTextInput}
  * @constructor
  */
 Blockly.FieldNumber = function(opt_value, opt_min, opt_max, opt_precision,
-    opt_validator) {
-  
+    opt_validator, opt_config) {
+
   /**
-   * The minimum value constraint.
+   * The minimum value this number field can contain.
    * @type {number}
    * @protected
    */
   this.min_ = -Infinity;
 
   /**
-   * The maximum value constraint.
+   * The maximum value this number field can contain.
    * @type {number}
    * @protected
    */
   this.max_ = Infinity;
 
   /**
-   * The precision constraint for the value.
+   * The multiple to which this fields value is rounded.
    * @type {number}
    * @protected
    */
   this.precision_ = 0;
 
+  /**
+   * The number of decimal places to allow, or null to allow any number of
+   * decimal digits.
+   * @type {?number}
+   * @protected
+   */
+  this.decimalPlaces_ = null;
+
   Blockly.FieldNumber.superClass_.constructor.call(
-      this, opt_value || 0, opt_validator);
+      this, opt_value || 0, opt_validator, opt_config);
 
   this.setConstraints(opt_min, opt_max, opt_precision);
```

Comment:
Do you think we can have this call the "internal" methods instead?

I'd hope that the value developers are passing to the constructor is within their constraints.

Reply:
I think this is fine.

Reply:
Sounds good, it does mean setValue gets called twice when the field is constructed though.